### ISSUE TITLE
Nitro plugins from project root are loaded before Nitro plugins in layers

### ISSUE THREAD

**Issue by Pierre-M** [nuxt/nuxt#23565]
### Environment

- Operating System: Darwin
- Node Version:     v18.14.0
- Nuxt Version:     3.7.4
- CLI Version:      3.9.0
- Nitro Version:    2.6.3
- Package Manager:  pnpm@7.0.0

### Reproduction

Here is a quick repro for the issue : 

* simple Nuxt project with one Nitro plugin in project root and another one in a local layer
* just added two console.log to showcase the loading order of those two plugins

https://stackblitz.com/edit/github-rrtlvf

### Describe the bug

When using Nuxt layers on a project, the Nitro plugins in the project root are loaded before the Nitro plugins in the layers.

That seems a little bit unlogic as layers are often use to share modules, configurations, start up settings between apps and therefore are meant to be executed first, before the app itself.

### Additional context

That issue has already been targeted for middlewares ordering : https://github.com/nuxt/nuxt/pull/22925.

I may be wrong but I think that make sense to follow the same logic on Nitro plugin ordering 🤗 

I think (but I may be wrong :) ) that the plugin ordering is defined here : 

https://github.com/nuxt/nuxt/blob/46be0767515eb862ab6ac660ce730e63e5b0283c/packages/nuxt/src/core/nitro.ts#L86C5-L86C12

For the middlewares order issue, the fix was juste to reverse the layers array, would that be a good solution for that one too ? 

### Logs

```shell-script
ℹ Vite client warmed up in 856ms                                                     18:12:44
✔ Nitro built in 417 ms                                                        nitro 18:12:44
Hello from app Nitro plugin !
Hello from layer Nitro plugin !
```

**Comment by Pierre-M** [comment]
Anything more I can do on that issue ? Btw thank you all for your amazing work, using Nuxt everyday is just delightful 🤗

**Comment by danielroe** [comment]
Yes, that makes sense to me! Feel free to submit a PR. We might generalise a utility to handle this kind of reordering - it's not quite as simple as reversing the order - see:

https://github.com/nuxt/nuxt/blob/89baee0493249719c3ae1552f83f4c3ceffacddb/packages/nuxt/src/core/app.ts#L200-L220
